db/revision_cache_lru.go

```diff
@@ -273,9 +263,10 @@ func (rc *LRURevisionCache) Upsert(ctx context.Context, docRev DocumentRevision)
 
 func (rc *LRURevisionCache) Invalidate(ctx context.Context, docID, revID string) {
 	value := rc.getValue(docID, revID, false)
-	if value != nil {
-		value.setInvalidFlag()
+	if value == nil {
+		return
 	}
+	rc.removeValue(value)
```

Comment:
`getValue` does a non-zero amount of work to update the LRU, even though we're about to remove the entry, as well as aquiring the revcache lock in both get and remove.

Maybe it's better to implement a remove method that only requires key?

You could essentially do:

```go
key := IDAndRev{DocID: docID, RevID: revID}
rc.lock.Lock()
defer rc.lock.Unlock()
if element := rc.cache[key]; element != nil {
	rc.lruList.Remove(element)
}
delete(rc.cache, key)
```